stacks_queues/queue.py

```diff
@@ -23,39 +23,88 @@ def enqueue(self, element):
             In the store are occupied
             returns None
         """
-        pass
+        if self.size == self.buffer_size:
+            raise QueueFullException ("Queue is full")
+
+        if self.size == 0:
+            self.front = 0
+            self.rear = 0
+
+        #per the test, don't duplicate the item if it was just enqueued
+        if self.store[(self.rear - 1 ) % self.buffer_size] != element:
+            self.store[self.rear] = element
+            self.rear = (self.rear + 1 ) % self.buffer_size
+            self.size = self.size + 1
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        #Check if empty, if so raise an exception
+        if self.size == 0:
+            raise QueueEmptyException("Queue is empty")
+
+        retVal = self.store[self.front]
+        #Find and store the front element
+        self.store[self.front] = None
+        #Move front to the next index
+        self.front = (self.front + 1 ) % self.buffer_size
+        # self.rear = (self.rear + 1 ) % self.buffer_size
+        self.size = self.size - 1    
+        return retVal    
+
+        
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+
+        return self.store[self.front]
         
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        return self.size
 
     def empty(self):
         """ Returns True if the Queue is empty
             And False otherwise.
         """
-        pass
+        if self.size == 0:
+            return True
+        else:
+            return False
```

Comment:
Small style suggestion
```suggestion
        return self.size == 0
```